Add editorial text on congestion control (#926)
moq-wg/moq-transport: draft-ietf-moq-transport.md

expects more OBJECTs to be delivered. The server closes the session with a
'GOAWAY Timeout' if the client doesn't close the session quickly enough.

## Congestion Control

MOQT does not specify a congestion controller, but there are important attributes
to consider when selecting a congestion controller for use with an application
built on top of MOQT.

### Bufferbloat

Traditional AIMD congestion controllers (ex. CUBIC {{?RFC9438}} and Reno {{?RFC6582}})

> are prone to Bufferbloat. Bufferbloat is when elements along the path build up
> a substantial queue or packets, commonly more than doubling the round trip time.

Reviewer: ```suggestion
are prone to Bufferbloat. Bufferbloat occurs when elements along the path build up
a substantial queue of packets, commonly more than doubling the round trip time.
```

These queued packets cause head-of-line blocking and latency, even when there is
no packet loss.

> ### Application-Limited

Reviewer 2: I feel like this session should probably also mention ABR somehow.

Author: Possibly so, but I don't have any text in mind at the moment.  PRs welcome, though.

The average bitrate for latency sensitive content needs to be less than the available

> bandwidth of the network, otherwise data will be queued and/or dropped. As such,

Author: ```suggestion
bandwidth, otherwise data will be queued and/or dropped. As such,
```

many MOQT applications will typically be limited by the available data to send, and
not the congestion controller. Many congestion control algorithms
only increase the congestion window or bandwidth estimate if fully utilized. This
combination can lead to underestimating the available network bandwidth. As a result,

> applications might need to ensure the congestion controller is not app-limited for at
> least a full round trip every so often to ensure the full available bandwidth can be

Author: ```suggestion
applications might need to periodically ensure the congestion controller is not
app-limited for at least a full round trip to ensure the available bandwidth can be
```

> measured.

Reviewer: Does MoQT need a mechanism to supply this data when the app is not (eg padding streams?)

Author: I'd rather not, because you can always request a higher quality at a low priority if there is one.

If there's not a higher quality, then you don't really need to discover more bandwidth.

Typically, I-Frames supply enough data to make the sender non app-limited for long enough to avoid completely under-utilizing the connection.

Senders control their congestion control, so client mechanisms to try to get their senders to unstick their congestion controllers are a bit scary.  This does remind me to read: https://datatracker.ietf.org/doc/draft-ietf-ccwg-ratelimited-increase/

Reviewer: See also #534

Reviewer 3: Tending to agree with @ianswett  .  Artificial load is always going to be very application specific and may or may not always help (from our BBR experimenations).
